alicloud/resource_alicloud_cdn_domain.go

```diff
@@ -0,0 +1,775 @@
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "on",
+							ValidateFunc: validateCdnEnable,
+						},
+					},
+				},
+				MaxItems: 1,
+			},
+			"auth_config": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"auth_type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "no_auth",
+							ValidateFunc: validateCdnAuthType,
+						},
+						"key1": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateCdnAuthKey,
+						},
+						"key2": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateCdnAuthKey,
+						},
+						"time_out": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "1800",
+						},
+					},
+				},
+				MaxItems: 1,
+			},
+			"http_header_configs": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"header_key": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateCdnHttpHeader,
+						},
+						"header_value": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				MaxItems: 10,
+			},
+			"cache_configs": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cache_content": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ttl": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"cache_type": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateCacheType,
+						},
+						"weight": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "1",
+							ValidateFunc: validateCacheWeight,
+						},
+						"id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAlicloudCdnDomainCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AliyunClient).cdnconn
+
+	args := cdn.AddDomainRequest{
+		DomainName: d.Get("domain_name").(string),
+		CdnType:    d.Get("cdn_type").(string),
+		SourcePort: d.Get("source_port").(int),
+		Scope:      d.Get("scope").(string),
```

Comment:
'scope' is not required and it doesn't have default, so there will be an error when scope is 'nil'.